src/core/InstrumentFunctions.cpp

```diff
@@ -464,18 +499,33 @@ void InstrumentFunctionArpeggio::processNote( NotePlayHandle * _n )
 		if( m_arpCycleModel.value() && dir != ArpDirection::Random )
 		{
 			cur_arp_idx *= m_arpCycleModel.value() + 1;
-			cur_arp_idx %= static_cast<int>( range / m_arpRepeatsModel.value() );
+			cur_arp_idx %= static_cast<int>((range + sortOffset) / m_arpRepeatsModel.value());
 		}
 
 		// If ArpDirection::Down or ArpDirection::DownAndUp, invert the final range.
 		if (dir == ArpDirection::Down || dir == ArpDirection::DownAndUp)
 		{
-			cur_arp_idx = static_cast<int>(range / m_arpRepeatsModel.value()) - cur_arp_idx - 1;
+			cur_arp_idx = static_cast<int>((range + sortOffset) / m_arpRepeatsModel.value()) - cur_arp_idx - 1;
 		}
 
 		// now calculate final key for our arp-note
-		const int sub_note_key = base_note_key + (cur_arp_idx / cur_chord_size ) *
+		int sub_note_key = 0;
+		if (static_cast<ArpMode>(m_arpModeModel.value()) != ArpMode::Sort)
+		{
+			sub_note_key = base_note_key + (cur_arp_idx / cur_chord_size) *
 							KeysPerOctave + chord_table.chords()[selected_arp][cur_arp_idx % cur_chord_size];
+		}
+		else
+		{
+			// if we are sorting, we already have the base note key and the chord key
+			sub_note_key = noteKeysArray[cur_arp_idx % total_chord_size] + (cur_arp_idx / total_chord_size) *
```

Comment:
Compiler is probably smart and will get the quotient and remainder at the same time, but just in case we may want to use `std::div` and for `cur_arp_idx` and `total_chord_size`? Similar for above block with `cur_chord_size` denominator.

Reply:
I think I do not fully understand this. Could you please describe it in an other way?

Reply:
Using `std::div` we pull the `%` and `/` operations out. Can make it easier to read. This isn't always needed since the operations are nearby and the compiler should be able to automatically do this for us, but it doesn't hurt readability and standardization.
```cpp
auto result = std::div(cur_arp_idx, total_chord_size);
sub_note_key = noteKeysArray[result.rem] + result.quot * ...;
```
Alternatively it could be done just by pulling the operations out:
```cpp
auto noteIndex = cur_arp_idx % total_chord_size;
auto offset = cur_arp_idx / total_chord_size;
sub_note_key = noteKeysArray[noteIndex] + offset * ...;
```

Reply:
Thanks for the explanation. I have implemented this.